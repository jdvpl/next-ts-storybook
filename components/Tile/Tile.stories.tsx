import { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { screen } from "@storybook/testing-library";
import { expect } from "@storybook/jest";

import Tile from "./Tile";

export default {
  title: "Content/Tile",
  component: Tile,
} as ComponentMeta<typeof Tile>;

export const BasicTile: ComponentStoryObj<typeof Tile> = {
  play: async ({ args }) => {
    await expect(screen.getByRole("heading")).toBeInTheDocument();
  },
  args: {
    header: "Hola Mundo",
    children:
      "orem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum",
    styles: "text-[5rem]",
  },
};
