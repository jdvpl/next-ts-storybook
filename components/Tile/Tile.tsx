import { FC } from "react";
import styled from "@emotion/styled";
import { borderRadius, boxShadow } from "@/components/styles";

type Props = {
  header: string;
  children?: string;
  styles?: string;
};
const Section = styled.section`
  ${({ theme }) =>
    boxShadow(theme.components.shadow1, theme.components.shadow2)};
  ${borderRadius};
  padding: 1vmin 4vmin 4vmin;
`;
const Tile: FC<Props> = ({ header, children, styles }) => {
  return (
    <Section>
      <h2 className={`text-[4rem] bg-slate-400`}>{header}</h2>
      {children}
    </Section>
  );
};

export default Tile;
