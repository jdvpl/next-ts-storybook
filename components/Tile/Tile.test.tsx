import React from "react";
import { render } from "@/test-utils";
import Tile from "./Tile";

describe("Tile test cases", () => {
  it("should render tile", () => {
    const { asFragment } = render(
      <Tile header="Titulo test">
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Magni
        consectetur officia provident tempora magnam? Dolor facilis dolorem,
        illo quia, debitis voluptatum deleniti eos expedita reiciendis minus
        nesciunt. Porro, eius expedita.
      </Tile>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
