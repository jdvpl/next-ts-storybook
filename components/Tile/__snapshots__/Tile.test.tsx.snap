// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile test cases should render tile 1`] = `
<DocumentFragment>
  <section
    class="css-1jpcu60"
  >
    <h2>
      Titulo test
    </h2>
    Lorem ipsum dolor sit amet, consectetur adipisicing elit. Magni consectetur officia provident tempora magnam? Dolor facilis dolorem, illo quia, debitis voluptatum deleniti eos expedita reiciendis minus nesciunt. Porro, eius expedita.
  </section>
</DocumentFragment>
`;
